pkg/parquetquery/iters.go

```diff
@@ -455,13 +456,24 @@ func NewSyncIterator(ctx context.Context, rgs []pq.RowGroup, column int, opts ..
 		curr:               EmptyRowNumber(),
 		at:                 IteratorResult{},
 		maxDefinitionLevel: MaxDefinitionLevel, // default value
+		clone:              true,               // default value
 	}
 
 	// Apply options
 	for _, opt := range opts {
 		opt(i)
 	}
 
+	// Always disable intern/clone for non-pointer types that don't need it.
+	// This eliminates unnecessary function calls on the hot path.
+	if len(rgs) > 0 {
+		cc := rgs[0].ColumnChunks()[column]
+		if cc.Type().Kind() != pq.ByteArray && cc.Type().Kind() != pq.FixedLenByteArray {
+			i.intern = false
```

Comment:
i know the intern logic wasn't added here, but it seems like we don't need an intern bool? we could just key off a non-nil intern pointer?

Reply:
Correct. Pushed some cleanup where instead of trying to figure things out at runtime in `makeResult()`, there are several variants and we "pre-compile" it here instead. Which allows for elimination of `intern bool` also.